Classes/URKArchive.mm

```diff
@@ -385,23 +395,75 @@ + (BOOL)urlIsARAR:(NSURL *)fileURL
     return [NSArray arrayWithArray:fileInfos];
 }
 
-- (nullable NSString *)firstVolumePath {
-    return @"";
+- (NSString *)firstVolumePath:(NSString *)filePath {
+    NSURL *fileURL = [self firstVolumeURL:[NSURL fileURLWithPath:filePath]];
+
+    return fileURL.path;
 }
 
-- (nullable NSURL *)firstVolumeURL {
-    NSString *path = [self firstVolumePath];
+- (NSURL *)firstVolumeURL:(NSURL *)fileURL {
     
-    if (![path length]) {
-        return nil;
+    URKLogDebug("Checking if the archive is part of a volume...");
+    
+    __block NSString *volumePath = fileURL.path;
```

Comment:
This doesn't need to be a `__block` variable, since it's not modified from inside a block. Also, once we refactor to a single class method (`+[URKArchive firstVolumePath:]`), it'll be the input argument.